### PR TITLE
fix: env driven approach for getting child gvr from parent gvk

### PR DESCRIPTION
**Pull request by prakash100198**
Making parent child gvk to gvr mapping config driven.
fixes:- https://github.com/devtron-labs/devtron/issues/3696

### CODE REVIEW

File: pkg/service/K8sService.go [hunk 1]
```diff
@@ -34,18 +36,87 @@ type K8sService interface {
...
+
+	gvkVsChildGvrAndScope := make(map[schema.GroupVersionKind][]k8sCommonBean.GvrAndScope)
+	k8sServiceImpl := &K8sServiceImpl{
+		logger:                logger,
+		helmReleaseConfig:     helmReleaseConfig,
+		gvkVsChildGvrAndScope: gvkVsChildGvrAndScope,
+	}
+	if len(helmReleaseConfig.ParentChildGvkMapping) > 0 {
+		err := k8sServiceImpl.populateParentChildGvkMapping(gvkVsChildGvrAndScope)
+		if err != nil {
```
Comment (nishant-d) [inline pkg/service/K8sService.go:69]: return and log error

File: pkg/service/K8sService.go [hunk 2]
```diff
@@ -34,18 +36,87 @@ type K8sService interface {
...
+func (impl K8sServiceImpl) populateParentChildGvkMapping(gvkVsChildGvrAndScope map[schema.GroupVersionKind][]k8sCommonBean.GvrAndScope) error {
+	var gvkChildMappings []ParentChildGvkMapping
+	parentChildGvkMapping := impl.helmReleaseConfig.ParentChildGvkMapping
+	err := json.Unmarshal([]byte(parentChildGvkMapping), &gvkChildMappings)
+	if err != nil {
+		impl.logger.Errorw("error in unmarshalling ParentChildGvkMapping", "parentChildGvkMapping", parentChildGvkMapping, "err", err)
+		return err
+	}
+	for _, parent := range gvkChildMappings {
+		childGvrAndScopes := make([]k8sCommonBean.GvrAndScope, 0)
```
Comment (nishant-d) [inline pkg/service/K8sService.go:85]: 1. make fix length array
2. use pointers
3. return value instead of caching directly

**Review by nishant-d — CHANGES_REQUESTED** [review]

File: pkg/service/K8sService.go (hunk 1 above)
Comment (prakash100198) [reply pkg/service/K8sService.go:70]: done

Comment (prakash100198) [reply pkg/service/K8sService.go:70]: done

File: pkg/service/K8sService.go (hunk 2 above)
Comment (prakash100198) [reply pkg/service/K8sService.go]: done

**Pull request merged**